src/components/web-form-renderer.vue

```diff
@@ -256,46 +264,138 @@ const getAttachment = (url) => {
   }).then(transformAttachmentResponse);
 };
 
-const postPrimaryInstance = (file) => {
+const postPrimaryInstance = async (file) => {
   let url = apiPaths.submissions(form.projectId, form.xmlFormId, !form.publishedAt, '');
 
   if (isEdit.value) {
     url = apiPaths.submission(form.projectId, form.xmlFormId, props.instanceId);
   }
   url = withToken(url);
-  return request({
-    method: isEdit.value ? 'PUT' : 'POST',
-    url,
-    data: file,
-    headers: {
-      'content-type': 'text/xml'
-    },
-    fulfillProblem: () => true
-  })
-    .then(({ data }) => {
-      if (isProblem(data)) {
-        if (data.code === 403.1) {
-          showModal(submissionModal, { type: 'sessionTimeoutModal' });
-        } else {
-          showModal(submissionModal, { type: 'errorModal', errorMessage: data.message });
-        }
-        return false;
-      }
-      return data.currentVersion.instanceId;
-    })
-    .catch(noop);
+  try {
+    const requestOptions = {
+      method: isEdit.value ? 'PUT' : 'POST',
+      url,
+      data: file,
+      headers: {
+        'content-type': 'text/xml'
+      },
+      alert: false,
+    };
+    const { data } = await request(requestOptions);
+    return { success: true, data };
+  } catch (error) {
+    return { success: false, data: error };
+  }
 };
 
 const uploadAttachment = async (attachment, instanceId) => {
-  const url = withToken(apiPaths.submissionAttachment(form.projectId, form.xmlFormId, !form.publishedAt, instanceId, attachment.file.name));
-  return request({
-    method: 'POST',
-    url,
-    data: attachment.file,
-    fulfillProblem: () => true
-  })
-    .then(({ data }) => data)
-    .catch(noop);
+  const url = withToken(apiPaths.submissionAttachment(form.projectId, form.xmlFormId, !form.publishedAt, instanceId, attachment.name));
+  const result = {};
+  try {
+    const requestOptions = {
+      method: 'POST',
+      url,
+      data: attachment,
+      alert: false,
+      headers: {
+        'content-type': attachment.type
+      }
+    };
+    const { data } = await request(requestOptions);
+    result.success = true;
+    result.data = data;
+  } catch (error) {
+    result.success = false;
+    result.data = error;
+  }
+
+  return { name: attachment.name, result };
```

Comment:
Does this need to return the name? The `attachmentResult` map already maps attachment name to result, so I feel like the name isn't needed here too. Not a big deal at all, might just be slightly simpler without it.

Reply:
it is needed to the value in `attachmentResult` map in this line: 

```
      submissionResult.attachmentResult.set(r.name, r.result);
```